client/rpcserver/handlers.go

```diff
@@ -143,133 +135,129 @@ func handleNewWallet(s *RPCServer, req *msgjson.Message) *msgjson.ResponsePayloa
 		errMsg := fmt.Sprintf("error creating %s wallet: %v",
 			dex.BipIDSymbol(form.AssetID), err)
 		resErr := msgjson.NewError(msgjson.RPCCreateWalletError, errMsg)
-		return createResponse(req.Route, nil, resErr)
+		return createResponse(newWalletRoute, nil, resErr)
 	}
 	s.notifyWalletUpdate(form.AssetID)
 	err = s.core.OpenWallet(form.AssetID, form.AppPass)
 	if err != nil {
 		errMsg := fmt.Sprintf("wallet connected, but failed to open with provided password: %v",
 			err)
 		resErr := msgjson.NewError(msgjson.RPCOpenWalletError, errMsg)
-		return createResponse(req.Route, nil, resErr)
+		return createResponse(newWalletRoute, nil, resErr)
 	}
 	s.notifyWalletUpdate(form.AssetID)
 	res := fmt.Sprintf(walletCreatedStr, dex.BipIDSymbol(form.AssetID))
-	return createResponse(req.Route, &res, nil)
+	return createResponse(newWalletRoute, &res, nil)
 }
 
 // handleOpenWallet handles requests for openWallet.
 // *msgjson.ResponsePayload.Error is empty if successful. Requires the app
 // password. Opens the wallet.
-func handleOpenWallet(s *RPCServer, req *msgjson.Message) *msgjson.ResponsePayload {
-	form := new(openWalletForm)
-	err := req.Unmarshal(form)
-	if err != nil {
-		resErr := msgjson.NewError(msgjson.RPCParseError,
-			"unable to unmarshal request")
-		return createResponse(req.Route, nil, resErr)
+func handleOpenWallet(s *RPCServer, params interface{}) *msgjson.ResponsePayload {
+	form, ok := params.(*openWalletForm)
+	if !ok {
+		panic("open wallet bad params")
 	}
-	err = s.core.OpenWallet(form.AssetID, form.AppPass)
+	err := s.core.OpenWallet(form.AssetID, form.AppPass)
 	if err != nil {
 		errMsg := fmt.Sprintf("error unlocking %s wallet: %v",
 			dex.BipIDSymbol(form.AssetID), err)
 		resErr := msgjson.NewError(msgjson.RPCOpenWalletError, errMsg)
-		return createResponse(req.Route, nil, resErr)
+		return createResponse(openWalletRoute, nil, resErr)
 	}
 	s.notifyWalletUpdate(form.AssetID)
 	res := fmt.Sprintf(walletUnlockedStr, dex.BipIDSymbol(form.AssetID))
-	return createResponse(req.Route, &res, nil)
+	return createResponse(openWalletRoute, &res, nil)
 }
 
 // handleCloseWallet handles requests for closeWallet.
 // *msgjson.ResponsePayload.Error is empty if successful. Closes the wallet.
-func handleCloseWallet(s *RPCServer, req *msgjson.Message) *msgjson.ResponsePayload {
-	var assetID uint32
-	err := req.Unmarshal(&assetID)
-	if err != nil {
-		resErr := msgjson.NewError(msgjson.RPCParseError,
-			"unable to unmarshal request")
-		return createResponse(req.Route, nil, resErr)
+func handleCloseWallet(s *RPCServer, params interface{}) *msgjson.ResponsePayload {
+	assetID, ok := params.(uint32)
+	fmt.Println(params)
+	if !ok {
+		panic("close wallet bad params")
```

Comment:
~~`fmt.Println(params)`~~

I hit this panic with a `./dexcctl closewallet 42`

```
Panic: close wallet bad params
goroutine 2066 [running]:
runtime/debug.Stack(0x1f, 0x0, 0x0)
	/usr/local/go/src/runtime/debug/stack.go:24 +0x9d
runtime/debug.PrintStack()
	/usr/local/go/src/runtime/debug/stack.go:16 +0x22
github.com/go-chi/chi/middleware.Recoverer.func1.1(0xc0003a5000, 0xeec700, 0xc000664540)
	/home/buck/go/pkg/mod/github.com/go-chi/chi@v4.0.2+incompatible/middleware/recoverer.go:28 +0x1b6
panic(0xb52bc0, 0xece1e0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
decred.org/dcrdex/client/rpcserver.handleCloseWallet(0xc000501360, 0xb51bc0, 0xc0012709a8, 0x0)
	/home/buck/programs/dcrdex/client/rpcserver/handlers.go:179 +0x3d7
decred.org/dcrdex/client/rpcserver.(*RPCServer).handleRequest(0xc000501360, 0xc000356640, 0x5)
	/home/buck/programs/dcrdex/client/rpcserver/rpcserver.go:388 +0x807
decred.org/dcrdex/client/rpcserver.(*RPCServer).parseHTTPRequest(0xc000501360, 0xeec700, 0xc000664540, 0xc000356640)
	/home/buck/programs/dcrdex/client/rpcserver/rpcserver.go:394 +0x3c
decred.org/dcrdex/client/rpcserver.(*RPCServer).handleJSON(0xc000501360, 0xeec700, 0xc000664540, 0xc0003a5000)
	/home/buck/programs/dcrdex/client/rpcserver/rpcserver.go:212 +0x376
net/http.HandlerFunc.ServeHTTP(0xc000133f10, 0xeec700, 0xc000664540, 0xc0003a5000)
	/usr/local/go/src/net/http/server.go:2007 +0x44
```

Reply:
your other suggestion will fix this